example-crates/example_cli/src/lib.rs

```diff
@@ -664,45 +666,213 @@ where
+                            Err(_) => true,
+                        })
+                        .collect::<Result<Vec<_>, _>>()?;
+
+                    for (spk_i, full_txo) in txouts {
+                        let addr = Address::from_script(&full_txo.txout.script_pubkey, network)?;
+                        println!(
+                            "{:?} {} {} {} spent:{:?}",
+                            spk_i, full_txo.txout.value, full_txo.outpoint, addr, full_txo.spent_by
+                        )
+                    }
+                    Ok(())
+                }
+            }
         }
         Commands::Send {
             value,
             address,
             coin_select,
+            chain_specfic,
         } => {
             let chain = &*chain.lock().unwrap();
             let address = address.require_network(network)?;
-            run_send_cmd(
-                graph,
-                db,
-                chain,
-                keymap,
-                coin_select,
-                address,
-                value,
-                broadcast,
-            )
+            let (transaction, change_index) = {
+                let graph = &mut *graph.lock().unwrap();
+                // take mutable ref to construct tx -- it is only open for a short time while building it.
+                let (tx, change_info) =
+                    create_tx(graph, chain, keymap, coin_select, address, value)?;
+
+                if let Some((index_changeset, (change_keychain, index))) = change_info {
+                    // We must first persist to disk the fact that we've got a new address from the
+                    // change keychain so future scans will find the tx we're about to broadcast.
+                    // If we're unable to persist this, then we don't want to broadcast.
+                    {
+                        let db = &mut *db.lock().unwrap();
+                        db.stage(C::from((
+                            local_chain::ChangeSet::default(),
+                            indexed_tx_graph::ChangeSet::from(index_changeset),
+                        )));
+                        db.commit()?;
+                    }
+
+                    // We don't want other callers/threads to use this address while we're using it
+                    // but we also don't want to scan the tx we just created because it's not
+                    // technically in the blockchain yet.
+                    graph.index.mark_used(&change_keychain, index);
+                    (tx, Some((change_keychain, index)))
+                } else {
+                    (tx, None)
+                }
+            };
+
+            match (broadcast)(chain_specfic, &transaction) {
+                Ok(_) => {
+                    println!("Broadcasted Tx : {}", transaction.txid());
+
+                    let keychain_changeset =
+                        graph.lock().unwrap().insert_tx(&transaction, None, None);
```

Comment:
Why does indexed tx graph `insert_tx` take three arguments but tx graph only take one. Eeek. We should make this the same. I like `TxGraph`'s API better FWIW.

Reply:
I guess the only benefit of having `insert_tx` also insert anchors and last_seen is we can reduce the number of times we calculate the txid :sweat_smile:.

I agree that the API between `TxGraph` and `IndexedTxGraph` should be relatively uniform. So `IndexedTxGraph` should have separate methods for inserting tx, anchor and last_seen.

Reply:
> I guess the only benefit of having `insert_tx` also insert anchors and last_seen is we can reduce the number of times we calculate the txid 😅.

How does it?

> However, I think this should be tackled in a separate issue.

Why? Does it need further discussion? I'd just do this now and forget about it.

Reply:
Done.